### PR TITLE
Prettification of new Index page

### PR DESCRIPTION
**Pull request by glasnt**
I've done some tabular data generation and formatting, using partials and css; and I've also created the 'outstanding unacked/unresolved' listings. 

A note on the outstanding listings - based on the timescale of the page (24 hours) I've only set the range to be the last day. Ideally it would be nice to get all records without a time_to_ack or time_to_resolve, but this [doesn't appear to be possible in influx](http://stackoverflow.com/questions/30091980/select-from-influxdb-where-value-is-null). So for now, I'm pulling all records for the last 24 hours and then moving them to an array if their values are null.

### CODE REVIEW

File: lib/influx.rb [hunk 1]
```diff
@@ -227,6 +227,85 @@ def threshold_recommendations(opts)
...
+          'entity' => entity,
+          'check'  => check,
+          'ack_by' => incident['acknowledge_by'],
+          'time_to_ack' => incident['time_to_ack'],
+          'time_to_resolve' => incident['time_to_resolve']
+        }
+        if member["time_to_ack"].nil? && member["time_to_resolve"].nil?
+ 	    unacked << member
+        elsif member["time_to_resolve"].nil?
+            puts incident
```
Comment (nerdyness) [inline lib/influx.rb]: debug output?

File: lib/influx.rb [hunk 2]
```diff
@@ -227,6 +227,84 @@ def threshold_recommendations(opts)
...
+        [
+          Chronic.parse("#{Date.today.strftime('%F')} #{x}:00:00 utc"),
+          Chronic.parse("#{Date.today.prev_day.strftime('%F')} #{x}:00:00 utc")
+        ]
+      }.flatten
+      shift_times.reject! { |x| x.to_i > Time.now.to_i }
+      shift_times = shift_times.push(Time.now.utc).sort.reverse[0..(shift_names.length + 1)]
+
+      shift_times.each_with_index { |time, index|
+        break if index + 1 == shift_times.length
```
Comment (nerdyness) [inline lib/influx.rb]: `break if time == shift_times.last` ?

File: lib/influx.rb [hunk 3]
```diff
@@ -68,7 +68,7 @@ def find_incidents(start_date = nil, end_date = nil, query_input = nil)
       end_date = Chronic.parse(end_date, :guess => false).last unless end_date.nil?
 
       # If we couldn't parse the given date, use the last 24 hours.
-      end_date = end_date.nil? ? Time.now.to_i : end_date.to_i
+      end_date = (end_date.nil? || end_date.to_i > Time.now.to_i) ? Time.now.to_i : end_date.to_i
```
Comment (nerdyness) [inline lib/influx.rb]: There is a lot happening in that line now. I'd be in favour of wrapping this over 2 lines so the code is more maintain and readable.

**Comment by nerdyness** [comment]
Nice work :+1: 
But wow there is a lot of complex code in there. Unfortunately I'm not sure how to write that any differently at the moment - maybe it's at it's best but it took me a while to understand `generate_daily_stats`.

**Comment by Hobbsee** [comment]
Yeah, i'm cleaning up generate_daily_stats to work with the config file, which includes a refactor.

**Comment by Hobbsee** [comment]
Lets get https://github.com/bulletproofnetworks/pigeonhole/pull/19 merged, while I work on the refactor of some of this.

**Pull request closed without merging**